environment.go

```diff
@@ -110,6 +109,53 @@ type Environment interface {
 	Close(context.Context) error
 }
 
+// Configure initializes the object. Some implementations may
+// not allow the same instance to be configured more than
+// once.
+//
+// If Configure returns without an error, you should assume
+// that the queues have been started, there was no issue
+// establishing a connection to the database, and that the
+// local and remote queues have started.
+//
+// Configure requires that either the path or DB is sent so that it
+// can construct the evergreen settings. If both are sent, the
+// settings will be from the file
+func NewEnvironment(ctx context.Context, confPath string, db *DBSettings) (Environment, error) {
```

Comment:
This should first validate that either db it not nil or path is non-empty.

Reply:
that's the first statement conditional statement in the current version. 

the fallback behavior here is pretty fiddly/delicate and basically every test fails if we change it a bit.